File: mocker-front/src/app/core/service/category.service.ts
import { Injectable } from '@angular/core';
import { CategoryModel } from '@core/domain/model/category.model';
import { AbstractService } from '@core/service/abstract.service';

/**
 * @author Do Quoc Viet
 */

@Injectable({
  providedIn: 'root'
})
export class CategoryService extends AbstractService<CategoryModel> {
  override ENTITY_URL = 'category';
}

File: mocker-back/src/main/java/com/mocker/service/CategoryService.java
package com.mocker.service;

import com.mocker.domain.exception.NotFoundException;
import com.mocker.domain.model.entity.Category;

import java.util.List;
import java.util.UUID;

/**
 * @author Luong Quoc Trung, Do Quoc Viet
 */

public interface CategoryService {

    /**
     * CATEGORY.01 GET /category
     *
     * @return categories
     */
    List<Category> getCategories();

    /**
     * CATEGORY.02 GET /category/{id}
     *
     * @param id of category
     * @return category
     */
    Category getCategoryHasGenerateTypes(UUID id) throws NotFoundException;

    /**
     * CATEGORY.03 GET /category/{id}/generate-types/count
     *
     * @param id of category
     * @return number of generate-types in category
     */
    Integer getGenerateTypesCount(UUID id) throws NotFoundException;

    List<Category> getCategoriesFetchGenerateTypes() throws NotFoundException;

}
